src/onnx/parse_slice.cpp

```diff
@@ -77,7 +86,43 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        if(reverse_direction == true)
+        {
+
+            migraphx::argument axes_arg = args.at(3)->eval();
+            std::vector<int> axes_v;
+            axes_arg.visit([&](auto s) { axes_v.assign(s.begin(), s.end()); });
+
+            auto lens = args[0]->get_shape().lens();
+
+            for(auto axis : axes_v)
+            {
+                auto start_v = op.starts[axis];
+                auto end_v   = op.ends[axis];
+                if((start_v < 0) & (end_v < INT_MIN))
+                {
+                    op.ends[axis]   = lens[axis] + start_v + 1;
+                    op.starts[axis] = 0;
+                }
+                else if((start_v < 0) & (end_v > INT_MIN) & (end_v < 0))
+                {
+                    op.ends[axis]   = lens[axis] + start_v + 1;
+                    op.starts[axis] = end_v - INT_MIN;
+                }
+            }
+
+            auto ins = info.add_instruction(op, args[0]);
+
+            for(auto axis : axes_v)
```

Comment:
Considering the cases here, I would suggest extend the reverse op to support multiple axes. We need the following changes:
1) Change the axis attribute in the reverse from int64_t axis to "std::vector<int64_t> axes"
2) For the ref compute function, change it to:

```
    argument compute(const shape& s, std::vector<argument> args) const
    {
        argument result{s};
        auto lens = s.lens();
        visit_all(result, args.front())([&](auto output, auto input) {
            shape_for_each(s, [&](const auto& out_idx) {
                auto in_idx              = out_idx;
                for (const auto& axis : axes)
                {
                    in_idx[axis] = lens[axis] - 1 - out_idx[axis];
                }
                output[s.index(out_idx)] = input[s.index(in_idx)];
            });
        });

        return result;
    }
```

For the device function, the implementation is:

```
argument reverse(hipStream_t stream, argument result, argument arg1, const std::vector<int64_t> axes)
{
    auto s       = arg1.get_shape();
    auto lens    = s.lens();
    std::size_t nelements = s.elements();
    visit_all(result, arg1)([&](auto output1, auto input1) {
        hip_visit_views(output1, input1, s)([&](auto output, auto input, auto hs) {
            gs_launch(stream, nelements)([=](auto i) __device__ {
                auto idx     = hs.multi(i);
                auto in_idx  = idx;
                for (auto axis : axes)
                {
                    in_idx[axis] = lens[axis] - 1 - idx[axis];
                }
                output[idx]  = input[in_idx];
            });
        });
    });

    return result;
}
```

Reply:
With supporting axes, we donot need the for loop here anymore.